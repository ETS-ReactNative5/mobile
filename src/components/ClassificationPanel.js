import React, { Component } from 'react';
import {
  Platform,
  TouchableOpacity,
  View
} from 'react-native';
import EStyleSheet from 'react-native-extended-stylesheet'
import StyledText from './StyledText'

const topPadding = (Platform.OS === 'ios') ? 10 : 0

class ClassificationPanel extends Component {
  render() {
    const tabs =
      <View style={styles.tabContainer}>
        <TouchableOpacity
          onPress={ () => { this.props.setQuestionVisibility(true) } }
          style={ this.props.isQuestionVisible ? [styles.tab] : [styles.tab, styles.deselectedTab] }>
          <StyledText text={ 'QUESTION' } />
        </TouchableOpacity>
        <TouchableOpacity
          onPress={ () => { this.props.setQuestionVisibility(false) } }
          style={ this.props.isQuestionVisible ? [styles.tab, styles.deselectedTab] : [styles.tab] }>
          <StyledText text={ 'TUTORIAL' } />
        </TouchableOpacity>
      </View>

    return (
      <View style={styles.container}>
        <View style={styles.panelContainer}>
          { this.props.hasTutorial ? tabs : null }
          { this.props.children }
        </View>
      </View>
    )
  }
}

const styles = EStyleSheet.create({
  $tabHeight: 32,
  $panelMargin: 13,
  container: {
    backgroundColor: '$lightestGrey',
    flex: 1,
    marginTop: 60,
    paddingTop: topPadding,
    paddingBottom: 0,
  },
  panelContainer: {
    flex: 1,
    backgroundColor: 'white',
    margin: '$panelMargin',
    marginBottom: 0,
  },
  tabContainer: {
    height: '$tabHeight',
    flexDirection: 'row',
    justifyContent: 'space-around',
    alignItems: 'center'
  },
  tab: {
    alignItems: 'center',
    justifyContent: 'center',
    height: '$tabHeight',
    width: '50% - $panelMargin',
    marginTop: 1,
  },
  deselectedTab: {
    backgroundColor: '$lightestGrey',
  }
})

ClassificationPanel.propTypes = {
  isFetching: React.PropTypes.bool,
  hasTutorial: React.PropTypes.bool,
  children: React.PropTypes.node,
  isQuestionVisible: React.PropTypes.bool,
  setQuestionVisibility: React.PropTypes.func,
}
export default ClassificationPanel
