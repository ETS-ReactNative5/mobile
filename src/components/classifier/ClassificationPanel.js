import React, { Component } from 'react';
import {
  TouchableOpacity,
  View
} from 'react-native';
import DeviceInfo from 'react-native-device-info'
import EStyleSheet from 'react-native-extended-stylesheet'
import PropTypes from 'prop-types';

import FontedText from '../common/FontedText'
import * as colorModes from '../../displayOptions/colorModes'

class ClassificationPanel extends Component {
  render() {
    const tabs =
      <View style={styles.tabContainer}>
        <TouchableOpacity
          onPress={ () => { this.props.setQuestionVisibility(true) } }
          style={ this.props.isQuestionVisible ?
              [styles.tab,  colorModes.contentBackgroundColorFor(this.props.inMuseumMode)] :
              [styles.tab,  colorModes.framingBackgroundColorFor(this.props.inMuseumMode)]
          }>
          <FontedText style={ this.props.isQuestionVisible ?
            [styles.tabText, colorModes.selectedTextColorFor(this.props.inMuseumMode)] :
            [styles.tabText, colorModes.selectedTextColorFor(this.props.inMuseumMode)]
          }>
            QUESTION
          </FontedText>
        </TouchableOpacity>
        <TouchableOpacity
          onPress={ () => { this.props.setQuestionVisibility(false) } }
          style={ this.props.isQuestionVisible ?
              [styles.tab,  colorModes.framingBackgroundColorFor(this.props.inMuseumMode)] :
              [styles.tab,  colorModes.contentBackgroundColorFor(this.props.inMuseumMode)]

          }>
          <FontedText style={ this.props.isQuestionVisible ?
            [styles.tabText, colorModes.selectedTextColorFor(this.props.inMuseumMode)]:
            [styles.tabText, colorModes.selectedTextColorFor(this.props.inMuseumMode)]
          }>
            TUTORIAL
          </FontedText>
        </TouchableOpacity>
      </View>

    return (
        <View style={[styles.panelContainer, this.props.containerStyle, colorModes.contentBackgroundColorFor(this.props.inMuseumMode)]}>
          { this.props.hasTutorial ? tabs : null }
          { this.props.children }
        </View>
    )
  }
}

const styles = EStyleSheet.create({
  panelContainer: {
    marginTop: 15,
    marginBottom: 0,
    marginHorizontal: 25
  },
  tabContainer: {
    flexDirection: 'row',
    justifyContent: 'space-around',
    alignItems: 'center'
  },
  tab: {
    alignItems: 'center',
    justifyContent: 'center',
    flex: 1,
    marginTop: 1,
  },
  tabText: {
    fontSize: DeviceInfo.isTablet() ? 22 : 14,
    marginVertical: 15
  }
})

ClassificationPanel.propTypes = {
  containerStyle: PropTypes.any,
  isFetching: PropTypes.bool,
  hasTutorial: PropTypes.bool,
  children: PropTypes.node,
  isQuestionVisible: PropTypes.bool,
  setQuestionVisibility: PropTypes.func,
  inMuseumMode: PropTypes.bool,
}
export default ClassificationPanel
