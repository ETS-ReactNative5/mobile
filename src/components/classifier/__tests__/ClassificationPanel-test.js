import 'react-native'
import React from 'react'
import renderer from 'react-test-renderer'
import ClassificationPanel from '../ClassificationPanel'

it('renders correctly', () => {
  const tree = renderer.create(
    <ClassificationPanel />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})
