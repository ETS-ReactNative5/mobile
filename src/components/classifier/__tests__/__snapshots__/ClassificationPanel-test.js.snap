// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={undefined}
>
  <View
    style={
      Array [
        undefined,
        undefined,
      ]
    }
  />
</View>
`;
